linux/keyman-system-service/build.sh

```diff
@@ -0,0 +1,66 @@
+#!/usr/bin/env bash
+
+## START STANDARD BUILD SCRIPT INCLUDE
+# adjust relative paths as necessary
+THIS_SCRIPT="$(readlink -f "${BASH_SOURCE[0]}")"
+. "${THIS_SCRIPT%/*}/../../resources/build/build-utils.sh"
+## END STANDARD BUILD SCRIPT INCLUDE
+
+################################ Main script ################################
+
+builder_describe \
+  "Build keyman-system-service." \
+  ":service" \
+  "clean" \
+  "configure" \
+  "build" \
+  "test" \
+  "install                   install artifacts" \
+  "uninstall                 uninstall artifacts"
+
+builder_parse "$@"
+
+cd "$THIS_SCRIPT_PATH/"
+
+if builder_is_debug_build; then
+  MESON_TARGET=debug
+  export CPPFLAGS=-DG_MESSAGES_DEBUG
+  export CFLAGS="-O0"
+  export CXXFLAGS="-O0"
+else
+  MESON_TARGET=release
+fi
+MESON_PATH="build/$(uname -m)/$MESON_TARGET"
+
+builder_describe_outputs \
+  configure "${MESON_PATH}/build.ninja" \
+  build "${MESON_PATH}/src/keyman-system-service"
+
+builder_run_action clean rm -rf "$THIS_SCRIPT_PATH/build/"
+
+# shellcheck disable=SC2086
+builder_run_action configure meson setup "$MESON_PATH" --werror --buildtype $MESON_TARGET "${builder_extra_params[@]}"
+
+if builder_start_action build; then
+  cd "$MESON_PATH"
+  ninja
+  builder_finish_action success build
+fi
+
+if builder_start_action test; then
+  cd "$MESON_PATH"
+  meson test --print-errorlogs $builder_verbose
+  builder_finish_action success test
+fi
+
+if builder_start_action install; then
+  cd "$MESON_PATH"
+  ninja install
+  builder_finish_action success install
+fi
+
+if builder_start_action uninstall; then
+  cd "$MESON_PATH"
+  ninja uninstall
+  builder_finish_action success uninstall
+fi
```

Comment:
You _could_ move `cd "$MESON_PATH"` out of the if blocks and then turn these into `builder_run_action` statements if you wanted to. But it's not important.

Reply:
Well, there's one complication: If these gets run as part of `linux/build.sh clean` it will fail because it can't cd into `$MESON_PATH` because that got deleted in clean action. 

The other a bit unfortunate thing is that if I run e.g. `linux/keyman-system-service/build.sh build` the output will also show the actions that come after the build action which is a bit confusing:

```
...
[linux/keyman-system-service] ## build starting...
[8/8] Linking target src/keyman-system-service
[linux/keyman-system-service] ## build completed successfully
builder_run_action test meson test --print-errorlogs
builder_run_action install ninja install
builder_run_action uninstall ninja uninstall
```

Reply:
> The other a bit unfortunate thing is that if I run e.g. linux/keyman-system-service/build.sh --debug build the output will also show the actions that come after the build action which is a bit confusing:

This has been fixed in f8535c469effd4b57fed0e3731f6f48d64ba8af6 (#8834)

Reply:
> Well, there's one complication: If these get run as part of linux/build.sh clean it will fail because it can't cd into $MESON_PATH because that got deleted in the clean action.

Ouch. Yeah, that's a shame... well, it looked good at the time!